### DISCUSSION TITLE
Page TOC navigation menu does not work with headers with custom ids

### DISCUSSION THREAD

**Discussion by adrian-martinez-vdshop** [requarks/wiki#4740] [Error / Bug Report]
Hi everyone, I'd like to discuss an issue I've just found. I am using Wiki.js 2.5.219, the latest version at the time of writing this.

# TL;DR
When trying to access through the TOC menu to a custom named header section, it does not work and throws the following error:
```
TypeError: Target must be a Number/Selector/HTMLElement/VueComponent, received undefined instead.
```

I'd expect that definitions like this one worked with TOC navigation:
```markdown
# Item A
## Description {#item-a-description}
(...)
```

# Full explanation and motivation

My use case is the following. I want to create a page with a variable (initially unknown) number of entries, ordered alphabetically for easy search, each of one having its own description:
```markdown
# Item A
## Description
---
# Item K
## Description
```
In the example above, it automatically generates anchors with ids `#description` and `#description-1`. It links it properly from TOC menu and it simply works. But when it comes to use this anchors to link to specific content from other pages, it gets unpractical to be used this way, as anchors automatic name are simply based on the order in which they appear in the page.

If I try to link to Item K description, following the example, I'd have to link to `my-page#description-1`. But once done, whenever item C gets inserted between Items A and K, previous link will target Item C instead of Item K, and Item K description header will have now `#description-2` as id.

So, for guaranteeing anchors uniqueness and invariability so links to it are not broken, I have two options, one being give explicitly a different name to every anchor, which seems impractical (and ugly, more when multiple header levels exist) too:
```markdown
# Item A
## Item A description
### Item A description subsection
#### Item A description subsection subsection
---
# Item K
## Item K description
### Item K description subsection
#### Item K description subsection subsection
```

The other alternative is way cleaner and seems to be already partially supported, by using [Heading IDs](https://www.markdownguide.org/extended-syntax/#heading-ids). The following notation is already supported and _almost_ works:
```markdown
# Item A
## Description {#item-a-description}
---
# Item K
## Description {#item-k-description}
---
(...)
```
It renders anchors with specified id, and when visiting `my-page#item-k-description`, it properly scrolls down smoothly to the appropriate section, which is nice when you are targeting specific content.
However, when trying to access through the TOC menu to a custom named anchors section, it does not work and shows the following error in the console:
```
TypeError: Target must be a Number/Selector/HTMLElement/VueComponent, received undefined instead.
```

This error makes TOC unusable for browsing page content, and makes it painful to get it to desired content, specially for large pages.

It would be so nice to get this fixed, thanks for all the great job you're doing !
